### DISCUSSION TITLE
Possibility of Filesystem dispatching events

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#35938] [Ideas]
Would be pretty awesome if `File::put($path, $contents)` dispatched an event:

```php
Event::listen('files.put: *', function($path, $contents) {

})
```

**Comment by shaedrich** [comment]
That would also allow for additional [test assertations](https://laravel.com/docs/testing) along with [`assertFileExists()`](https://docs.phpunit.de/en/9.6/assertions.html#assertfileexists):
* `\Illuminate\Support\Facade\File::assertFileCreated()`
* `\Illuminate\Support\Facade\File::assertFileDeleted()`
* `\Illuminate\Support\Facade\File::assertFileUpdated()`
* `\Illuminate\Support\Facade\File::assertFileMoved()`
* `\Illuminate\Support\Facade\File::assertFileAppended()`
* `\Illuminate\Support\Facade\File::assertFilePrepended()`
* etc.
